docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,90 @@
+# Teleport Architecture
```

Comment:
I don't like using Gramine debug mode. Remote attestation etc is unavailable. I would rather just not use Gramine when running outside enclave